NPE with new json deserialization support of non EntityView but classic DTO or shuch is provided

If you use a normal DTO like https://github.com/EugenMayer/java-spring-pocs/blob/master/src/main/java/de/kontextwork/poc/spring/blaze/fullapp/subject/group/model/api/TestDTO.java

```java
@Data
@Builder
public class TestDTO
{
  String name;
}
```

if you deserialize this, you get an NPE 

```
Caused by: java.lang.NullPointerException
	at com.blazebit.persistence.integration.jackson.EntityViewAwareObjectMapper.readerFor(EntityViewAwareObjectMapper.java:71)
	at com.blazebit.persistence.spring.data.rest.impl.json.EntityViewAwareMappingJackson2HttpMessageConverter.readJavaType(EntityViewAwareMappingJackson2HttpMessageConverter.java:71)
	at com.blazebit.persistence.spring.data.rest.impl.json.EntityViewAwareMappingJackson2HttpMessageConverter.read(EntityViewAwareMappingJackson2HttpMessageConverter.java:59)
	at org.springframework.web.servlet.mvc.method.annotation.AbstractMessageConverterMethodArgumentResolver.readWithMessageConverters(AbstractMessageConverterMethodArgumentResolver.java:205)
	at org.springframework.web.servlet.mvc.method.annotation.RequestResponseBodyMethodProcessor.readWithMessageConverters(RequestResponseBodyMethodProcessor.java:158)
	at org.springframework.web.servlet.mvc.method.annotation.RequestResponseBodyMethodProcessor.resolveArgument(RequestResponseBodyMethodProcessor.java:131)
	at org.springframework.web.method.support.HandlerMethodArgumentResolverComposite.resolveArgument(HandlerMethodArgumentResolverComposite.java:127)
	at org.springframework.web.method.support.InvocableHandlerMethod.getMethodArgumentValues(InvocableHandlerMethod.java:167)
	at org.springframework.web.method.support.InvocableHandlerMethod.invokeForRequest(InvocableHandlerMethod.java:134)
	at org.springframework.web.servlet.mvc.method.annotation.ServletInvocableHandlerMethod.invokeAndHandle(ServletInvocableHandlerMethod.java:105)
	at org.springframework.web.servlet.mvc.method.annotation.RequestMappingHandlerAdapter.invokeHandlerMethod(RequestMappingHandlerAdapter.java:893)
	at org.springframework.web.servlet.mvc.method.annotation.RequestMappingHandlerAdapter.handleInternal(RequestMappingHandlerAdapter.java:798)
	at org.springframework.web.servlet.mvc.method.AbstractHandlerMethodAdapter.handle(AbstractHandlerMethodAdapter.java:87)
	at org.springframework.web.servlet.DispatcherServlet.doDispatch(DispatcherServlet.java:1040)
	at org.springframework.web.servlet.DispatcherServlet.doService(DispatcherServlet.java:943)
	at org.springframework.web.servlet.FrameworkServlet.processRequest(FrameworkServlet.java:1006)
	... 100 more
```
from com.blazebit.persistence.integration.jackson.EntityViewAwareObjectMapper#canRead

where 

```
ManagedViewType<?> managedViewType = entityViewManager.getMetamodel().managedView(type);
```

`managedViewType` is null, while the `type` is a java list
Seems like the BP integration, out of any reason, thinks it is a EV? ( `TestDTO` are used in a list here as a payload


## Reproduction

As usual at 
https://github.com/EugenMayer/java-spring-pocs/blob/master/src/test/java/de/kontextwork/poc/spring/blaze/fullapp/subject/group/controller/GroupControllerTest.java#L134

--- a/integration/jackson/pom.xml
+++ b/integration/jackson/pom.xml
@@ -37,6 +37,32 @@
             <version>7.0</version>
             <scope>provided</scope>
         </dependency>
+        <dependency>
+            <groupId>${project.groupId}</groupId>
+            <artifactId>blaze-persistence-core-impl</artifactId>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>${project.groupId}</groupId>
+            <artifactId>blaze-persistence-entity-view-impl</artifactId>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.hibernate</groupId>
+            <artifactId>hibernate-entitymanager</artifactId>
+            <version>${version.hibernate-5.4}</version>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>${project.groupId}</groupId>
+            <artifactId>blaze-persistence-integration-hibernate-5.4</artifactId>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>junit</groupId>
+            <artifactId>junit</artifactId>
+            <scope>test</scope>
+        </dependency>
     </dependencies>
 
     <build>

--- a/integration/jackson/src/main/java/com/blazebit/persistence/integration/jackson/EntityViewAwareObjectMapper.java
+++ b/integration/jackson/src/main/java/com/blazebit/persistence/integration/jackson/EntityViewAwareObjectMapper.java
@@ -19,17 +19,18 @@ package com.blazebit.persistence.integration.jackson;
 import com.blazebit.persistence.view.EntityViewManager;
 import com.blazebit.persistence.view.metamodel.ManagedViewType;
 import com.blazebit.persistence.view.metamodel.ViewMetamodel;
-import com.blazebit.persistence.view.metamodel.ViewType;
 import com.fasterxml.jackson.annotation.JsonAutoDetect;
 import com.fasterxml.jackson.annotation.PropertyAccessor;
 import com.fasterxml.jackson.databind.BeanDescription;
 import com.fasterxml.jackson.databind.DeserializationConfig;
-import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.JavaType;
 import com.fasterxml.jackson.databind.JsonDeserializer;
 import com.fasterxml.jackson.databind.ObjectMapper;
 import com.fasterxml.jackson.databind.ObjectReader;
 import com.fasterxml.jackson.databind.deser.BeanDeserializerModifier;
 import com.fasterxml.jackson.databind.module.SimpleModule;
+
+import java.util.Collection;
 
 /**
  * @author Christian Beikov
@@ -48,7 +49,7 @@ public class EntityViewAwareObjectMapper {
         module.setDeserializerModifier(new BeanDeserializerModifier() {
             @Override
             public JsonDeserializer<?> modifyDeserializer(DeserializationConfig config, BeanDescription beanDesc, JsonDeserializer<?> deserializer) {
-                ViewType<?> view = metamodel.view(beanDesc.getBeanClass());
+                ManagedViewType<?> view = metamodel.managedView(beanDesc.getBeanClass());
                 if (view != null) {
                     return new EntityViewReferenceDeserializer(entityViewManager, view, (JsonDeserializer<Object>) deserializer);
                 }
@@ -57,22 +58,37 @@ public class EntityViewAwareObjectMapper {
         });
         objectMapper.registerModule(module);
         objectMapper.setVisibility(PropertyAccessor.GETTER, JsonAutoDetect.Visibility.NONE);
-        objectMapper.disable(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES);
 
         this.objectMapper = objectMapper;
     }
 
+    public ObjectMapper getObjectMapper() {
+        return objectMapper;
+    }
+
     public boolean canRead(Class<?> clazz) {
-        return entityViewManager.getMetamodel().view(clazz) != null;
+        return entityViewManager.getMetamodel().managedView(clazz) != null;
+    }
+
+    public boolean canRead(JavaType javaType) {
+        if (!javaType.isContainerType()) {
+            return canRead(javaType.getRawClass());
+        } else if (javaType.isCollectionLikeType()) {
+            return canRead(javaType.getContentType().getRawClass());
+        }
+        return false;
+    }
+
+    public ObjectReader readerFor(JavaType javaType) {
+        if (Collection.class.isAssignableFrom(javaType.getRawClass())) {
+            return objectMapper.readerFor(javaType);
+        } else {
+            return readerFor(javaType.getRawClass());
+        }
     }
 
     public ObjectReader readerFor(Class<?> type) {
-        ManagedViewType<?> managedViewType = entityViewManager.getMetamodel().managedView(type);
-        if (managedViewType.isCreatable()) {
-            Object createView = entityViewManager.create(managedViewType.getJavaType());
-            return objectMapper.readerForUpdating(createView);
-        }
-        return objectMapper.readerFor(managedViewType.getJavaType());
+        return objectMapper.readerFor(type);
     }
 
 }

--- a/integration/jackson/src/main/java/com/blazebit/persistence/integration/jackson/EntityViewReferenceDeserializer.java
+++ b/integration/jackson/src/main/java/com/blazebit/persistence/integration/jackson/EntityViewReferenceDeserializer.java
@@ -17,14 +17,15 @@
 package com.blazebit.persistence.integration.jackson;
 
 import com.blazebit.persistence.view.EntityViewManager;
+import com.blazebit.persistence.view.metamodel.ManagedViewType;
 import com.blazebit.persistence.view.metamodel.MethodAttribute;
 import com.blazebit.persistence.view.metamodel.ViewType;
 import com.fasterxml.jackson.core.JsonParser;
-import com.fasterxml.jackson.core.JsonProcessingException;
 import com.fasterxml.jackson.core.ObjectCodec;
 import com.fasterxml.jackson.databind.DeserializationContext;
 import com.fasterxml.jackson.databind.JsonDeserializer;
 import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
 
 import java.io.IOException;
 import java.util.HashMap;
@@ -64,73 +65,61 @@ public class EntityViewReferenceDeserializer extends JsonDeserializer {
     private final String idAttribute;
     private final IdType idType;
     private final boolean updatable;
+    private final boolean creatable;
 
-    public EntityViewReferenceDeserializer(EntityViewManager entityViewManager, ViewType<?> view, JsonDeserializer<Object> defaultDeserializer) {
+    public EntityViewReferenceDeserializer(EntityViewManager entityViewManager, ManagedViewType<?> view, JsonDeserializer<Object> defaultDeserializer) {
         this.entityViewManager = entityViewManager;
         this.defaultDeserializer = defaultDeserializer;
         this.entityViewClass = view.getJavaType();
-        MethodAttribute<?, ?> idAttribute = view.getIdAttribute();
-        this.idAttribute = idAttribute.getName();
-        IdType idType = ID_TYPES.get(idAttribute.getJavaType());
-        if (idType == null) {
-            throw new IllegalArgumentException("Can't create entity view reference deserializer for entity view '" + entityViewClass.getName() + "' because id attribute '" + idAttribute.getName() + "' has an unsupported id type: " + idAttribute.getJavaType().getName());
+        if (view instanceof ViewType<?>) {
+            MethodAttribute<?, ?> idAttribute = ((ViewType<?>) view).getIdAttribute();
+            this.idAttribute = idAttribute.getName();
+            IdType idType = ID_TYPES.get(idAttribute.getJavaType());
+            if (idType == null) {
+                throw new IllegalArgumentException("Can't create entity view reference deserializer for entity view '" + entityViewClass.getName() + "' because id attribute '" + idAttribute.getName() + "' has an unsupported id type: " + idAttribute.getJavaType().getName());
+            }
+            this.idType = idType;
+        } else {
+            this.idAttribute = null;
+            this.idType = null;
         }
-        this.idType = idType;
         this.updatable = view.isUpdatable();
+        this.creatable = view.isCreatable();
     }
 
     @Override
-    public Object deserialize(JsonParser jsonParser, DeserializationContext deserializationContext) throws IOException, JsonProcessingException {
+    public Object deserialize(JsonParser jsonParser, DeserializationContext deserializationContext) throws IOException {
         Object reference = null;
         ObjectCodec codec = jsonParser.getCodec();
         JsonNode treeNode = codec.readTree(jsonParser);
-        JsonNode jsonNode = treeNode.get(idAttribute);
-        Object id;
-        if (jsonNode != null && !jsonNode.isNull()) {
-            switch (idType) {
-                case BYTE:
-                    id = jsonNode.numberValue().byteValue();
-                    break;
-                case SHORT:
-                    id = jsonNode.numberValue().shortValue();
-                    break;
-                case INTEGER:
-                    id = jsonNode.numberValue().intValue();
-                    break;
-                case LONG:
-                    id = jsonNode.numberValue().longValue();
-                    break;
-                case FLOAT:
-                    id = jsonNode.numberValue().floatValue();
-                    break;
-                case DOUBLE:
-                    id = jsonNode.numberValue().doubleValue();
-                    break;
-                case CHARACTER:
-                    String s = jsonNode.textValue();
-                    if (s.length() != 1) {
-                        throw new IllegalArgumentException("Expected id value to be exactly one character but was: " + s);
-                    }
-                    id = s.charAt(0);
-                    break;
-                case STRING:
-                    id = jsonNode.textValue();
-                    break;
-                default:
-                    id = null;
-                    break;
-            }
 
-            reference = entityViewManager.getReference(entityViewClass, id);
-            if (updatable) {
-                jsonParser = codec.treeAsTokens(treeNode);
-                jsonParser.nextToken();
-                return deserializationContext.findNonContextualValueDeserializer(deserializationContext.constructType(reference.getClass()))
-                        .deserialize(jsonParser, deserializationContext, reference);
+        // We create also creatable & updatable views if no id is given
+        // If an id is given in such a case, we create a reference for updates
+        if (creatable && (!updatable || treeNode.get(idAttribute) == null)) {
+            reference = entityViewManager.create(entityViewClass);
+        } else if (idAttribute != null) {
+            JsonNode jsonNode = treeNode.get(idAttribute);
+            Object id;
+            if (jsonNode != null && !jsonNode.isNull()) {
+                if (idType == null) {
+                    id = null;
+                } else {
+                    id = idType.read(jsonNode);
+                }
+                // Remove the id attribute since we deserialized it already
+                ((ObjectNode) treeNode).without(idAttribute);
+                reference = entityViewManager.getReference(entityViewClass, id);
             }
         }
 
-        return reference;
+        if (reference == null) {
+            return null;
+        }
+
+        jsonParser = codec.treeAsTokens(treeNode);
+        jsonParser.nextToken();
+        return deserializationContext.findNonContextualValueDeserializer(deserializationContext.constructType(reference.getClass()))
+                .deserialize(jsonParser, deserializationContext, reference);
     }
 
     /**
@@ -138,13 +127,60 @@ public class EntityViewReferenceDeserializer extends JsonDeserializer {
      * @since 1.4.0
      */
     enum IdType {
-        BYTE,
-        SHORT,
-        INTEGER,
-        LONG,
-        FLOAT,
-        DOUBLE,
-        CHARACTER,
-        STRING
+        BYTE {
+            @Override
+            Object read(JsonNode jsonNode) {
+                return jsonNode.numberValue().byteValue();
+            }
+        },
+        SHORT {
+            @Override
+            Object read(JsonNode jsonNode) {
+                return jsonNode.numberValue().shortValue();
+            }
+        },
+        INTEGER {
+            @Override
+            Object read(JsonNode jsonNode) {
+                return jsonNode.numberValue().intValue();
+            }
+        },
+        LONG {
+            @Override
+            Object read(JsonNode jsonNode) {
+                return jsonNode.numberValue().longValue();
+            }
+        },
+        FLOAT {
+            @Override
+            Object read(JsonNode jsonNode) {
+                return jsonNode.numberValue().floatValue();
+            }
+        },
+        DOUBLE {
+            @Override
+            Object read(JsonNode jsonNode) {
+                return jsonNode.numberValue().doubleValue();
+            }
+        },
+        CHARACTER {
+            @Override
+            Object read(JsonNode jsonNode) {
+
+                String s = jsonNode.textValue();
+                if (s.length() != 1) {
+                    throw new IllegalArgumentException("Expected id value to be exactly one character but was: " + s);
+                }
+                return s.charAt(0);
+            }
+        },
+        STRING {
+            @Override
+            Object read(JsonNode jsonNode) {
+                return jsonNode.textValue();
+            }
+        };
+
+        abstract Object read(JsonNode jsonNode);
     }
 }

--- a/integration/jackson/src/test/java/com/blazebit/persistence/integration/jackson/EntityViewAwareObjectMapperTest.java
+++ b/integration/jackson/src/test/java/com/blazebit/persistence/integration/jackson/EntityViewAwareObjectMapperTest.java
@@ -1,0 +1,236 @@
+/*
+ * Copyright 2014 - 2019 Blazebit.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.blazebit.persistence.integration.jackson;
+
+import com.blazebit.persistence.Criteria;
+import com.blazebit.persistence.CriteriaBuilderFactory;
+import com.blazebit.persistence.view.CreatableEntityView;
+import com.blazebit.persistence.view.EntityView;
+import com.blazebit.persistence.view.EntityViews;
+import com.blazebit.persistence.view.IdMapping;
+import com.blazebit.persistence.view.UpdatableEntityView;
+import com.blazebit.persistence.view.spi.EntityViewConfiguration;
+import com.blazebit.persistence.view.spi.type.EntityViewProxy;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.ObjectReader;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import javax.persistence.EntityManagerFactory;
+import javax.persistence.Persistence;
+import java.lang.reflect.ParameterizedType;
+import java.lang.reflect.Type;
+import java.util.List;
+
+/**
+ * @author Christian Beikov
+ * @since 1.4.0
+ */
+public class EntityViewAwareObjectMapperTest {
+
+    static EntityManagerFactory emf;
+    static CriteriaBuilderFactory cbf;
+
+    @BeforeClass
+    public static void prepare() {
+        emf = Persistence.createEntityManagerFactory("Test");
+        cbf = Criteria.getDefault().createCriteriaBuilderFactory(emf);
+    }
+
+    static EntityViewAwareObjectMapper mapper(Class<?>... classes) {
+        EntityViewConfiguration configuration = EntityViews.createDefaultConfiguration();
+        for (Class<?> clazz : classes) {
+            configuration.addEntityView(clazz);
+        }
+
+        ObjectMapper objectMapper = new ObjectMapper();
+        return new EntityViewAwareObjectMapper(configuration.createEntityViewManager(cbf), objectMapper);
+    }
+
+    @Test
+    public void testReadView() throws Exception {
+        EntityViewAwareObjectMapper mapper = mapper(NameView.class);
+        ObjectReader objectReader = mapper.readerFor(mapper.getObjectMapper().constructType(NameView.class));
+        NameView view = objectReader.readValue("{\"id\": 1}");
+        Assert.assertEquals(1L, view.getId());
+        Assert.assertNull(view.getName());
+    }
+
+    @Test
+    public void testReadViewList() throws Exception {
+        EntityViewAwareObjectMapper mapper = mapper(NameView.class);
+        ObjectReader objectReader = mapper.readerFor(mapper.getObjectMapper().constructType(new ParameterizedType() {
+            @Override
+            public Type[] getActualTypeArguments() {
+                return new Type[]{ NameView.class };
+            }
+
+            @Override
+            public Type getRawType() {
+                return List.class;
+            }
+
+            @Override
+            public Type getOwnerType() {
+                return null;
+            }
+        }));
+        List<NameView> views = objectReader.readValue("[{\"id\": 1}, {\"id\": 2}]");
+        Assert.assertEquals(2, views.size());
+        Assert.assertEquals(1L, views.get(0).getId());
+        Assert.assertNull(views.get(0).getName());
+        Assert.assertEquals(2L, views.get(1).getId());
+        Assert.assertNull(views.get(0).getName());
+    }
+
+    @EntityView(SomeEntity.class)
+    interface NameView {
+        @IdMapping
+        long getId();
+        String getName();
+    }
+
+    @Test
+    public void testReadViewWithSetters() throws Exception {
+        EntityViewAwareObjectMapper mapper = mapper(ReadViewWithSetters.class, NameView.class);
+        ObjectReader objectReader = mapper.readerFor(mapper.getObjectMapper().constructType(ReadViewWithSetters.class));
+        ReadViewWithSetters view = objectReader.readValue("{\"id\": 1, \"name\": \"test\", \"parent\": {\"id\": 2}}");
+        Assert.assertEquals(1L, view.getId());
+        Assert.assertEquals("test", view.getName());
+        Assert.assertEquals(2L, view.getParent().getId());
+    }
+
+    @EntityView(SomeEntity.class)
+    interface ReadViewWithSetters {
+        @IdMapping
+        long getId();
+        String getName();
+        void setName(String name);
+        NameView getParent();
+        void setParent(NameView parent);
+    }
+
+    @Test
+    public void testUpdatableView() throws Exception {
+        EntityViewAwareObjectMapper mapper = mapper(UpdateViewWithSetters.class, NameView.class);
+        ObjectReader objectReader = mapper.readerFor(mapper.getObjectMapper().constructType(UpdateViewWithSetters.class));
+        UpdateViewWithSetters view = objectReader.readValue("{\"id\": 1, \"name\": \"test\", \"parent\": {\"id\": 2}}");
+        Assert.assertEquals(1L, view.getId());
+        Assert.assertEquals("test", view.getName());
+        Assert.assertEquals(2L, view.getParent().getId());
+    }
+
+    @EntityView(SomeEntity.class)
+    @UpdatableEntityView
+    interface UpdateViewWithSetters {
+        @IdMapping
+        long getId();
+        String getName();
+        void setName(String name);
+        NameView getParent();
+        void setParent(NameView parent);
+    }
+
+    @Test
+    public void testCreatableView() throws Exception {
+        EntityViewAwareObjectMapper mapper = mapper(CreatableViewWithSetters.class, NameView.class);
+        ObjectReader objectReader = mapper.readerFor(mapper.getObjectMapper().constructType(CreatableViewWithSetters.class));
+        CreatableViewWithSetters view = objectReader.readValue("{\"name\": \"test\", \"parent\": {\"id\": 2}}");
+        Assert.assertTrue(((EntityViewProxy) view).$$_isNew());
+        Assert.assertEquals("test", view.getName());
+        Assert.assertEquals(2L, view.getParent().getId());
+    }
+
+    @EntityView(SomeEntity.class)
+    @CreatableEntityView
+    interface CreatableViewWithSetters {
+        @IdMapping
+        long getId();
+        String getName();
+        void setName(String name);
+        NameView getParent();
+        void setParent(NameView parent);
+    }
+
+    @Test
+    public void testCreatableAndUpdatableView() throws Exception {
+        EntityViewAwareObjectMapper mapper = mapper(CreatableAndUpdatableViewWithSetters.class, NameView.class);
+        ObjectReader objectReader = mapper.readerFor(mapper.getObjectMapper().constructType(CreatableAndUpdatableViewWithSetters.class));
+        CreatableAndUpdatableViewWithSetters view = objectReader.readValue("{\"id\": 1, \"name\": \"test\", \"parent\": {\"id\": 2}}");
+        Assert.assertFalse(((EntityViewProxy) view).$$_isNew());
+        Assert.assertEquals(1L, view.getId());
+        Assert.assertEquals("test", view.getName());
+        Assert.assertEquals(2L, view.getParent().getId());
+    }
+
+    @Test
+    public void testCreatableAndUpdatableViewWithoutId() throws Exception {
+        EntityViewAwareObjectMapper mapper = mapper(CreatableAndUpdatableViewWithSetters.class, NameView.class);
+        ObjectReader objectReader = mapper.readerFor(mapper.getObjectMapper().constructType(CreatableAndUpdatableViewWithSetters.class));
+        CreatableAndUpdatableViewWithSetters view = objectReader.readValue("{\"name\": \"test\", \"parent\": {\"id\": 2}}");
+        Assert.assertTrue(((EntityViewProxy) view).$$_isNew());
+        Assert.assertEquals("test", view.getName());
+        Assert.assertEquals(2L, view.getParent().getId());
+    }
+
+    @EntityView(SomeEntity.class)
+    @CreatableEntityView
+    @UpdatableEntityView
+    interface CreatableAndUpdatableViewWithSetters {
+        @IdMapping
+        long getId();
+        String getName();
+        void setName(String name);
+        NameView getParent();
+        void setParent(NameView parent);
+    }
+
+    @Test
+    public void testCreatableAndUpdatableViewWithoutNestedUpdatableView() throws Exception {
+        EntityViewAwareObjectMapper mapper = mapper(CreatableAndUpdatableViewWithNested.class, CreatableAndUpdatableViewWithSetters.class, NameView.class);
+        ObjectReader objectReader = mapper.readerFor(mapper.getObjectMapper().constructType(CreatableAndUpdatableViewWithNested.class));
+        CreatableAndUpdatableViewWithNested view = objectReader.readValue("{\"name\": \"test\", \"parent\": {\"id\": 2, \"name\": \"parent\"}}");
+        Assert.assertTrue(((EntityViewProxy) view).$$_isNew());
+        Assert.assertEquals("test", view.getName());
+        Assert.assertEquals(2L, view.getParent().getId());
+        Assert.assertEquals("parent", view.getParent().getName());
+    }
+
+    @Test
+    public void testCreatableAndUpdatableViewWithoutNestedUpdatableViewWithoutId() throws Exception {
+        EntityViewAwareObjectMapper mapper = mapper(CreatableAndUpdatableViewWithNested.class, CreatableAndUpdatableViewWithSetters.class, NameView.class);
+        ObjectReader objectReader = mapper.readerFor(mapper.getObjectMapper().constructType(CreatableAndUpdatableViewWithNested.class));
+        CreatableAndUpdatableViewWithNested view = objectReader.readValue("{\"name\": \"test\", \"parent\": {\"name\": \"parent\"}}");
+        Assert.assertTrue(((EntityViewProxy) view).$$_isNew());
+        Assert.assertEquals("test", view.getName());
+        Assert.assertTrue(((EntityViewProxy) view.getParent()).$$_isNew());
+        Assert.assertEquals("parent", view.getParent().getName());
+    }
+
+    @EntityView(SomeEntity.class)
+    @CreatableEntityView
+    @UpdatableEntityView
+    interface CreatableAndUpdatableViewWithNested {
+        @IdMapping
+        long getId();
+        String getName();
+        void setName(String name);
+        CreatableAndUpdatableViewWithSetters getParent();
+        void setParent(CreatableAndUpdatableViewWithSetters parent);
+    }
+}

--- a/integration/jackson/src/test/java/com/blazebit/persistence/integration/jackson/SomeEntity.java
+++ b/integration/jackson/src/test/java/com/blazebit/persistence/integration/jackson/SomeEntity.java
@@ -1,0 +1,39 @@
+/*
+ * Copyright 2014 - 2019 Blazebit.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.blazebit.persistence.integration.jackson;
+
+import javax.persistence.Entity;
+import javax.persistence.FetchType;
+import javax.persistence.Id;
+import javax.persistence.ManyToOne;
+import javax.persistence.OneToMany;
+import java.util.Set;
+
+/**
+ * @author Christian Beikov
+ * @since 1.4.0
+ */
+@Entity
+public class SomeEntity {
+    @Id
+    Long id;
+    String name;
+    @ManyToOne(fetch = FetchType.LAZY)
+    SomeEntity parent;
+    @OneToMany(mappedBy = "parent")
+    Set<SomeEntity> children;
+}

--- a/integration/jackson/src/test/resources/META-INF/persistence.xml
+++ b/integration/jackson/src/test/resources/META-INF/persistence.xml
@@ -1,0 +1,24 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+  Copyright 2014 - 2019 Blazebit.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+  -->
+<persistence version="2.0" xmlns="http://java.sun.com/xml/ns/persistence" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://java.sun.com/xml/ns/persistence http://java.sun.com/xml/ns/persistence/persistence_2_0.xsd">
+    <persistence-unit name="Test" transaction-type="RESOURCE_LOCAL">
+        <class>com.blazebit.persistence.integration.jackson.SomeEntity</class>
+        <properties>
+            <property name="hibernate.dialect" value="org.hibernate.dialect.H2Dialect"/>
+        </properties>
+    </persistence-unit>
+</persistence>

--- a/integration/spring-data/rest/src/main/java/com/blazebit/persistence/spring/data/rest/impl/json/EntityViewAwareMappingJackson2HttpMessageConverter.java
+++ b/integration/spring-data/rest/src/main/java/com/blazebit/persistence/spring/data/rest/impl/json/EntityViewAwareMappingJackson2HttpMessageConverter.java
@@ -49,7 +49,21 @@ public class EntityViewAwareMappingJackson2HttpMessageConverter extends MappingJ
     }
 
     @Override
+    public boolean canRead(Type type, Class<?> contextClass, MediaType mediaType) {
+        JavaType javaType = getJavaType(type, contextClass);
+        if (!entityViewAwareObjectMapper.canRead(javaType)) {
+            return false;
+        }
+        return super.canRead(type, contextClass, mediaType);
+    }
+
+    @Override
     public boolean canWrite(Class<?> clazz, MediaType mediaType) {
+        return false;
+    }
+
+    @Override
+    public boolean canWrite(Type type, Class<?> clazz, MediaType mediaType) {
         return false;
     }
 
@@ -68,7 +82,7 @@ public class EntityViewAwareMappingJackson2HttpMessageConverter extends MappingJ
                             readValue(inputMessage.getBody());
                 }
             }
-            return entityViewAwareObjectMapper.readerFor(javaType.getRawClass()).readValue(inputMessage.getBody());
+            return entityViewAwareObjectMapper.readerFor(javaType).readValue(inputMessage.getBody());
         } catch (IOException ex) {
             throw new HttpMessageNotReadableException("Could not read document: " + ex.getMessage(), ex);
         }

